### PR TITLE
EUI-6179: mock task priorisation

### PR DESCRIPTION
**Pull request by Surekha-HMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-6179

### Change description ###

Added mock search task
Added Urgent priority and badge
Default sort based on priority
If sorted by another column, priority sort becomes secondary 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts [hunk 1]
```diff
@@ -85,7 +85,7 @@ export class AllWorkTaskComponent extends TaskListWrapperComponent {
       };
       return {
         search_parameters: searchParameters,
-        sorting_parameters: [this.getSortParameter()],
+        sorting_parameters: this.getSortParameter(),
```
Comment (connorpgpmcelroy) [inline src/work-allocation-2/containers/all-work-task/all-work-task.component.ts:88]: Minor query - can I just have a quick comment on why this change is required please?

File: src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts [hunk 2]
```diff
@@ -133,7 +133,7 @@ export class TaskListWrapperComponent implements OnDestroy, OnInit {
     return {
       service: TaskService.IAC,
       defaultSortDirection: SortOrder.ASC,
-      defaultSortFieldName: this.getDateField('dueDate'),
+      defaultSortFieldName: 'priority',
```
Comment (connorpgpmcelroy) [inline src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts:136]: Same as previous comment about constant variable - ignore if to do with mocks

File: src/work-allocation-2/enums/priority-limits.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export enum PriorityLimits {
+    Urgent= 2000,
+    High= 5000,
```
Comment (connorpgpmcelroy) [inline src/work-allocation-2/enums/priority-limits.ts:3]: Minor comment - spacing of equals is off here (see below enum)

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Number of changes/queries

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts (hunk 1 above)
Comment (DavidJayakumar) [reply src/work-allocation-2/containers/all-work-task/all-work-task.component.ts:88]: getSortParameter() returns an array or sortParameter. So this is required

File: src/work-allocation-2/components/priority-field/priority-field.component.spec.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { TaskPriority } from '../../enums';
 import { PriorityFieldComponent } from './priority-field.component';
 
-describe('PriorityFieldComponent', () => {
+fdescribe('PriorityFieldComponent', () => {
```
[the remark this answers is not in the document]
Comment (DavidJayakumar) [reply src/work-allocation-2/components/priority-field/priority-field.component.spec.ts:6]: Removed

[the remark this answers is not in the document]
Comment (DavidJayakumar) [reply src/work-allocation-2/components/priority-field/priority-field.component.spec.ts:6]: fixed

File: src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts (hunk 2 above)
Comment (DavidJayakumar) [reply src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts:136]: Default sort order is changed from dueDate to priority. Yes it will be handled in mock based on value passed

File: src/work-allocation-2/enums/priority-limits.ts (hunk 3 above)
Comment (DavidJayakumar) [reply src/work-allocation-2/enums/priority-limits.ts:3]: Done

File: api/workAllocation2/index.ts [hunk 5]
```diff
@@ -138,13 +138,20 @@ export async function searchTask(req: EnhancedRequest, res: Response, next: Next
     const basePath: string = prepareSearchTaskUrl(baseWorkAllocationTaskUrl);
     const postTaskPath = preparePaginationUrl(req, basePath);
     const searchRequest = req.body.searchRequest;
+    let prioritySortParameter;
+    searchRequest.sorting_parameters.find((sort, index) => {
+      if(sort.sort_by === 'priority') {
```
[the remark this answers is not in the document]
Comment (DavidJayakumar) [reply api/workAllocation2/index.ts:143]: Yes it is mock - temporary code and will be removed once api get the functionality to handle this

[the remark this answers is not in the document]
Comment (connorpgpmcelroy) [reply api/workAllocation2/index.ts:143]: Great

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts (hunk 1 above)
Comment (connorpgpmcelroy) [reply src/work-allocation-2/containers/all-work-task/all-work-task.component.ts:88]: Fair enough, was just intrigued as to why it was like this before - LGTM

File: src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts (hunk 2 above)
Comment (connorpgpmcelroy) [reply src/work-allocation-2/containers/task-list-wrapper/task-list-wrapper.component.ts:136]: Great

File: src/cases/components/case-task/case-task.component.ts [hunk 6]
```diff
@@ -2,6 +2,7 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
 import { Router } from '@angular/router';
 import { AlertService } from '@hmcts/ccd-case-ui-toolkit';
 import { InfoMessage } from '../../../work-allocation-2/enums';
+import { PriorityLimits } from '../../../work-allocation-2/enums';
```
Comment (connorpgpmcelroy) [inline src/cases/components/case-task/case-task.component.ts:5]: Imports could be grouped together

File: src/work-allocation-2/enums/index.ts [hunk 7]
```diff
@@ -11,6 +11,7 @@ import { TaskPriority } from './task-priority';
 import { TaskService } from './task-service';
 import { TaskSort } from './task-sort';
 import { TaskView } from './task-view';
+import { PriorityLimits } from './priority-limits';
```
Comment (connorpgpmcelroy) [inline src/work-allocation-2/enums/index.ts:14]: Import ordering wrong here - 'p' before 't'

**Review by connorpgpmcelroy — APPROVED** [review]
Approved with minor comments

**Review by udaydenduluri33 — APPROVED** [review]
@DavidJayakumar / @Surekha-HMCTS  Thanks for the changes

**Review by Mo-Lala84 — APPROVED** [review]
approved
